[Task4, Sprint2] [Team 4] Github actions to run serverless artillery tests

Performance Test
**Using GitHub actions to run serverless artillery tests against the /users API when new YAML is checked into the development repository**

**Serverless Artillery**
Using GitHub actions to run serverless artillery tests against the /users API when new YAML is checked into the development repository.

1.  Install serverless artillery on Local Machine https://medium.com/serverlessguru/load-testing-serverless-applications-with-serverless-artillery-1ab2597f463e
    a)Installing serverless-artillery
        npm install -g serverless-artillery
2.To check that the installation succeeded, run:
          slsart --version
3. Download Project Folder in VSC
4.  Create folders for Workflows and Testing in VSC
5. Setting up the Load Test Configuration (script.yml)
6. Deploy to AWS
         slsart deploy --stage <your-unique-stage-name>
       Start the load Test
        slsart invoke --stage <your-unique-stage-name>
8. Create a YML file in Workflows folder named "serverless_art_github-actions.yml" in VSC
9. In script.yml file(which is in the Serverless-Artillery folder) write in the necessary code in VSC
9. Create a text file in the Testing folder named "testing.txt" in VSC
10. Implement code in "serverless_art_github-actions.yml"
11. Add Secrets AWS keys to github.
11. Create a mock testing code in the  "testing.txt"file just to do testing in your repo 
before pushing it the NSC repo
12. Save all files in VSC
13. Upstage Changes in VSC
14. Commit your changes and message of what you are trying to accomplish
15. Push your code to the origin
16. On GitHub, navigate to the main page of the repository.
18. Under your repository name, click Actions.
19. In the left sidebar, click the workflow you want to see.
20. Under Jobs or in the visualization graph, click the job you want to see.
21. View the results of each step.
22. Hopefully there will be no errors, However my PT has an Authorization error
23. Create a Pull request and Assign a group member to review it .
##**##Error that I am receiving###**
**An error occurred:  at IncomingMessage.emit (events.js:327:22)
      at IncomingMessage.EventEmitter.emit (domain.js:467:12)
      at endReadableNT (internal/streams/readable.js:1327:12)
      at processTicksAndRejections (internal/process/task_queues.js:80:21) {
    code: 'AccessDenied',
    time: 2021-02-02T06:40:05.025Z,
    requestId: '06c0b436-9585-4a63-b17a-77d615886df8',
    statusCode: 403,
    retryable: false,
    retryDelay: 44.203190115775314
  }
}
Error: Process completed with exit code 1..**

|Date|Activity|TimeSpent|
|-|-|-|
|1/24/2021|Group Meeting via Zoom|1 hour|
|1/27/2021|Created environment in VSC to start testing |1 hour|
|1/31/2021| Github actions trying to fix errors for Deploying, invoke, remove fuctions  | 6 hours ||
|1/1/2021| Trying to figure Error code: 'AccessDenied',Error: Process completed with exit code 1. | 6 hours|

--- a/.github/workflows/serverless_art_github-actions.yml
+++ b/.github/workflows/serverless_art_github-actions.yml
@@ -1,7 +1,7 @@
 name: Performance-Test
 
 on:  [push]
- # push:
+  #push:
    # branches:
     #  - development
 

--- a/.github/workflows/serverless_art_github-actions.yml
+++ b/.github/workflows/serverless_art_github-actions.yml
@@ -1,9 +1,9 @@
 name: Performance-Test
 
-on:  
-  push:
-    branches:
-      - development
+on:  [push]
+ # push:
+   # branches:
+    #  - development
 
 jobs:
   slsart-perf:

--- a/testing/Serverless-Artillery/users_api_script.yml
+++ b/testing/Serverless-Artillery/users_api_script.yml
@@ -2,7 +2,7 @@
 # This default script is intended to get you started quickly.
 # There is a lot more that Artillery can do.
 # You can find great documentation of the possibilities at:
-# https://artillery.io/docs/  ####
+# https://artillery.io/docs/  ########
 config:
   # this hostname will be used as a prefix for each URI in the flow unless a complete URI is specified
   target: "https://nsc-fun-dev-usw2-tsmtestfunction.azurewebsites.net/api/HttpTriggerLoadTest"

--- a/testing/Serverless-Artillery/users_api_script.yml
+++ b/testing/Serverless-Artillery/users_api_script.yml
@@ -2,7 +2,7 @@
 # This default script is intended to get you started quickly.
 # There is a lot more that Artillery can do.
 # You can find great documentation of the possibilities at:
-# https://artillery.io/docs/
+# https://artillery.io/docs/  ####
 config:
   # this hostname will be used as a prefix for each URI in the flow unless a complete URI is specified
   target: "https://nsc-fun-dev-usw2-tsmtestfunction.azurewebsites.net/api/HttpTriggerLoadTest"

